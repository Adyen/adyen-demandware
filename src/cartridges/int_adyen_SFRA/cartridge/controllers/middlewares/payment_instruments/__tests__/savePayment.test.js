/* eslint-disable global-require */
let savePayment;
let res;
let req;

beforeEach(() => {
  const { paymentInstruments } = require('../../index');
  savePayment = paymentInstruments.savePayment;
  jest.clearAllMocks();
  res = { json: jest.fn() };
  req = { currentCustomer: { profile: { customerNo: 'mocked_customerNo' } } };
});

afterEach(() => {
  jest.resetModules();
});

describe('Save Payment', () => {
  it('should do nothing if adyen secured fields is not enabled', () => {
    const AdyenHelper = require('*/cartridge/scripts/util/adyenHelper');
    const server = require('server');
    AdyenHelper.getAdyenSecuredFieldsEnabled.mockImplementation(() => false);
    savePayment.call({ emit: jest.fn() }, req, res, jest.fn());
    expect(server.forms.getForm).toBeCalledTimes(0);
  });

  it('should fail if zeroAuth has error', () => {
    const adyenZeroAuth = require('*/cartridge/scripts/adyenZeroAuth');
    adyenZeroAuth.zeroAuthPayment.mockImplementation(() => ({
      error: true,
    }));
    savePayment.call({ emit: jest.fn() }, req, res, jest.fn());
    expect(res.json.mock.calls).toMatchSnapshot();
  });

  it('should fail if resultCode is not Authorised', () => {
    const adyenZeroAuth = require('*/cartridge/scripts/adyenZeroAuth');
    adyenZeroAuth.zeroAuthPayment.mockImplementation(() => ({
      resultCode: 'Not_Authorised',
    }));
    savePayment.call({ emit: jest.fn() }, req, res, jest.fn());
    expect(res.json.mock.calls).toMatchSnapshot();
  });

  it('should succeed', () => {
    savePayment.call({ emit: jest.fn() }, req, res, jest.fn());
    expect(res.json.mock.calls).toMatchSnapshot();
  });

  it('should return redirectAction and succeed', () => {
    const adyenZeroAuth = require('*/cartridge/scripts/adyenZeroAuth');
    adyenZeroAuth.zeroAuthPayment.mockImplementation(() => ({
      resultCode: 'RedirectShopper',
      action: {
        paymentMethodType: "scheme",
        url: "https://checkoutshopper-test.adyen.com/checkoutshopper/threeDS2.shtml",
        data: {
          MD: "mockMD",
          PaReq: "mockPaReq",
          TermUrl: "https://checkoutshopper-test.adyen.com/checkoutshopMock"},
        method: "POST",
        type: "redirect"
      }
    }));
    savePayment.call({ emit: jest.fn() }, req, res, jest.fn());
    expect(res.json.mock.calls).toMatchSnapshot();
  });
});
