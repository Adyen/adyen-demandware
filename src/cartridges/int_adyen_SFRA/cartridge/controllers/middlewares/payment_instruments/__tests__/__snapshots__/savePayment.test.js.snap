// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Save Payment should fail if resultCode is not Authorised 1`] = `
Array [
  Array [
    Object {
      "error": Array [
        "mocked_error.card.information.error",
      ],
      "success": false,
    },
  ],
]
`;

exports[`Save Payment should fail if zeroAuth has error 1`] = `
Array [
  Array [
    Object {
      "error": Array [
        "mocked_error.card.information.error",
      ],
      "success": false,
    },
  ],
]
`;

exports[`Save Payment should succeed 1`] = `
Array [
  Array [
    Object {
      "redirectUrl": "[\\"PaymentInstruments-List\\"]",
      "success": true,
    },
  ],
]
`;

exports[`Save Payment should return redirectAction and succeed 1`] = `
Array [
  Array [
    Object {
      "redirectUrl": "[\\"PaymentInstruments-List\\"]",
      redirectAction: {
        paymentMethodType: "scheme",
        url: "https://checkoutshopper-test.adyen.com/checkoutshopper/threeDS2.shtml",
        data: {
            MD: "mockMD",
            PaReq: "mockPaReq",
            TermUrl: "https://checkoutshopper-test.adyen.com/checkoutshopMock"},
        method: "POST",
        type: "redirect"
      },
      "success": true,
    },
  ],
]
`;
